service/controller/azure_machine_pool.go

```diff
@@ -54,7 +55,7 @@ type AzureMachinePoolConfig struct {
 	OIDC                      setting.OIDC
 	RegistryDomain            string
 	SentryDSN                 string
-	SSHUserList               string
+	SSHUserList               []v1alpha1.ClusterKubernetesSSHUser
```

Comment:
Can't we keep the type to `string`? I think there is value in having `azure_machine_pool` controller not depending on `v1alpha1` types, and only use go types or CAPI/CAPZ types.

Reply:
we can, at the cost of duplicating the parsing function in two places (or moving it into a shared location).
What about having a new local type to hold this information?

Reply:
Duplication is fine sometimes, specially having in mind that `AzureConfig` related stuff will be removed in the following months. Anyway, your call.

Reply:
I created an external module for parsing the flag and pass around a new custom type between controllers.
WDYT?